backend/src/main/kotlin/app/ehrenamtskarte/backend/verification/database/repos/CardRepository.kt

```diff
@@ -21,9 +21,18 @@ object CardRepository {
         return if (query == null) null else CardEntity.wrapRow(query)
     }
 
-    fun insert(cardInfoHash: ByteArray, totpSecret: ByteArray?, expirationDay: Long?, regionId: Int, issuerId: Int, codeType: CodeType) =
+    fun findByHashModelAndActivationSecret(project: String, hashModel: ByteArray, activationSecret: ByteArray): CardEntity? {
+        val query = (Projects innerJoin Regions innerJoin Cards)
+            .slice(Cards.columns)
+            .select { Projects.project eq project and (Cards.cardInfoHash eq hashModel) and (Cards.activationSecret eq activationSecret) }
+            .singleOrNull()
+        return if (query == null) null else CardEntity.wrapRow(query)
+    }
+
+    fun insert(cardInfoHash: ByteArray, activationSecret: ByteArray?, totpSecret: ByteArray?, expirationDay: Long?, regionId: Int, issuerId: Int, codeType: CodeType) =
```

Comment:
The totpSecret should always be null when inserting a new card, so we can omit the argument.